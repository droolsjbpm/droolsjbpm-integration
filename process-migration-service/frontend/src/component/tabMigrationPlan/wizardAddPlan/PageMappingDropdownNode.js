import React from "react";

import { DropdownButton, MenuItem } from "patternfly-react";

export default class PageMappingDropdownNode extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: this.props.title
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(option) {
    const newTitle = this.props.title + ":" + option;
    this.setState({ title: newTitle });
    this.props.onDropdownChange(option);
  }

  createMenuItems(options) {
    let menuItems = [];
    for (var i = 0; i < options.length; i++) {
      const value = options[i].value;
      const label = options[i].label;
      menuItems.push(
        <MenuItem key={i} eventKey={value} onSelect={this.handleChange}>
          {label}
        </MenuItem>
      );
    }
    return menuItems;
  }

  render() {
    //const defaultOption = this.state.selected;
    //const placeHolderValue = typeof this.state.selected === 'string' ? this.state.selected : this.state.selected.value

    //const title = 'Primary';

    return (
      <div>
        <DropdownButton title={this.state.title} id="PageMappingDropdownButton">
          {this.createMenuItems(this.props.options)}
        </DropdownButton>
      </div>
    );
  }
}
