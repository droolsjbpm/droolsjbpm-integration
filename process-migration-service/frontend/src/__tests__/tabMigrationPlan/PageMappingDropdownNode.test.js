import renderer from "react-test-renderer";
import React from "react";
import PageMappingDropdownNode from "../../component/tabMigrationPlan/wizardAddPlan/PageMappingDropdownNode";

test("PageMappingDropdownNode renders correctly using snapshot", () => {
  const sourceNode = [
    {
      value: "_D3E17247-1D94-47D8-93AD-D645E317B736",
      label: "Self Evaluation:_D3E17247-1D94-47D8-93AD-D645E317B736"
    },
    {
      value: "_E35438DF-03AF-4D7B-9DCB-30BC70E7E92E",
      label: "PM Evaluation:_E35438DF-03AF-4D7B-9DCB-30BC70E7E92E"
    },
    {
      value: "_AB431E82-86BC-460F-9D8B-7A7617565B36",
      label: "HR Evaluation:_AB431E82-86BC-460F-9D8B-7A7617565B36"
    },
    {
      value: "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E",
      label: "Input:_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E"
    }
  ];
  const myMock = jest.fn();
  const tree = renderer
    .create(
      <PageMappingDropdownNode
        options={sourceNode}
        title="Source Nodes"
        onDropdownChange={myMock}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
