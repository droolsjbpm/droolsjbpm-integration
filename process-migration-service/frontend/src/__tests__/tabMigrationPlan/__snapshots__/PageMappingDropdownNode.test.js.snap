// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMappingDropdownNode renders correctly using snapshot 1`] = `
<div>
  <div
    className="dropdown btn-group"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-default"
      disabled={false}
      id="PageMappingDropdownButton"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      type="button"
    >
      Source Nodes
       
      <span
        className="caret"
      />
    </button>
    <ul
      aria-labelledby="PageMappingDropdownButton"
      className="dropdown-menu"
      role="menu"
    >
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          Self Evaluation:_D3E17247-1D94-47D8-93AD-D645E317B736
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          PM Evaluation:_E35438DF-03AF-4D7B-9DCB-30BC70E7E92E
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          HR Evaluation:_AB431E82-86BC-460F-9D8B-7A7617565B36
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          Input:_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E
        </a>
      </li>
    </ul>
  </div>
</div>
`;
